plugins/inputs/redis_sentinel/README.md

```diff
@@ -0,0 +1,215 @@
+# Telegraf Plugin: Redis Sentinel
+
+A plugin for Redis Sentinel to monitor multiple Sentinel instances that are
+mointoring multiple Redis servers and replicas.
+
+### Configuration:
+
+```
+# Read Redis Sentinel's basic status information
+[[inputs.redis_sentinel]]
+  ## specify servers via a url matching:
+  ##  [protocol://][:password]@address[:port]
+  ##  e.g.
+  ##    tcp://localhost:26379
+  ##    tcp://:password@192.168.99.100
+  ##
+  ## If no servers are specified, then localhost is used as the host.
+  ## If no port is specified, 26379 is used
+  servers = ["tcp://localhost:26379"]
+
+  ## specify server password
+  # password = "s#cr@t%"
+
+  ## Optional TLS Config
+  # tls_ca = "/etc/telegraf/ca.pem"
+  # tls_cert = "/etc/telegraf/cert.pem"
+  # tls_key = "/etc/telegraf/key.pem"
+  ## Use TLS but skip chain & host verification
+  # insecure_skip_verify = true
+```
+
+### Measurements & Fields:
+
+The plugin gathers the results of these commands and measurements:
+
+* `sentinel masters` - `redis_sentinel_masters`
+* `sentinel sentinels` - `redis_sentinels`
+* `sentinel replicas` - `redis_replicas`
+* `info all` - `redis_sentinel`
+
+The `has-quorum` (int) field in `redis_sentinel_masters` is from calling the command `sentinels ckquorum`. `1` is for true, `0` is for false.
+
+There are 5 remote network requests made for each server listed in the config.
+
+
+### Metrics
+
+- redis_sentinel_masters
+  - tags:
+    - host
+    - master_name
+    - port
+    - source
+
+  - fields:
+    - config_epoch (int)
+    - down_after_milliseconds (int)
+    - failover_timeout (int)
+    - flags (string)
+    - has_quorum (int)
+    - info_refresh (int)
+    - ip (string)
+    - last_ok_ping_reply (int)
+    - last_ping_reply (int)
+    - last_ping_sent (int)
+    - link_pending_commands (int)
+    - link_refcount (int)
+    - name (string)
+    - num_other_sentinels (int)
+    - num_slaves (int)
+    - parallel_syncs (int)
+    - port (int)
+    - quorum (int)
+    - role_reported (string)
+    - role_reported_time (int)
+    - runid (string)
+
+- redis_sentinels
+  - tags:
+    - host
+    - master_name
+    - port
+    - sentinel_ip
+    - sentinel_port
+    - source
+
+  - fields:
+    - down_after_milliseconds (int)
+    - flags (string)
+    - ip (string)
+    - last_hello_message (int)
+    - last_ok_ping_reply (int)
+    - last_ping_reply (int)
+    - last_ping_sent (int)
+    - link_pending_commands (int)
+    - link_refcount (int)
+    - name (string)
+    - port (int)
+    - runid (string)
```

Comment:
I think we should skip runid, this is a random value that is recalculated each time you start redis.